src/GSoft.Extensions.MediatR.Analyzers/SemanticDesignAnalyzer.cs

```diff
@@ -50,41 +50,31 @@ private static void OnCompilationStarted(CompilationStartAnalysisContext context
 
     private sealed class AnalyzerImplementation
     {
-        private readonly HashSet<INamedTypeSymbol> _mediatorTypesWithSendMethod;
-        private readonly HashSet<INamedTypeSymbol> _requestHandlerTypes;
-
-        public AnalyzerImplementation(Compilation compilation)
+        private static readonly HashSet<string> MediatorSendAndSendAsyncMethodNames = new HashSet<string>(StringComparer.Ordinal)
```

Comment:
We don't want this one as ImmutableHashSet too?

Reply:
We don't need to, because it's "kind of a constant" that is executed a single time for the whole compilation.
When it comes Microsoft's built-in analyzers, they do use [HashSet<>](https://github.com/search?q=repo%3Adotnet%2Froslyn-analyzers+%22new+HashSet%3Cstring%3E%22&type=code) but only for strings.

Whenever they compare multiple symbols at runtime, they use ImmutableHashSet[Builder].